tests/test_view_model.py

```diff
@@ -0,0 +1,59 @@
+from datetime import date, timedelta
+from todo_app.data.view_model import ViewModel
+from todo_app.data.todo_item import TodoItem
+
+def test_view_to_do_model():
+   # assert 
+   todo = TodoItem("1234", "To Do", "Test Title", date.today)
+   doing = TodoItem("1234", "Doing", "Test Title", date.today)
+   items=[todo, doing]
+   view_model = ViewModel(items=items)
+   
+   assert len(view_model.todo_items) == 1
+   assert view_model.todo_items[0].status == "To Do"
+   
+
+def test_view_doing_model():
+   # assert 
+   doing = TodoItem("5678", "Doing", "Test Title", date.today)
+   items=[doing]
+   view_model = ViewModel(items=items)
+   
+   assert len(view_model.doing_items) == 1
+   assert view_model.doing_items[0].status == "Doing"
+
+   
+def test_view_done_model():
+   # assert 
+   done = TodoItem("9876", "Done", "Test Title", date.today)
+   items=[done]
+   view_model = ViewModel(items=items)
+   
+   assert len(view_model.done_items) == 1
+   assert view_model.done_items[0].status == "Done"
+
+def test_view_done_model_less_than_five():
+   # assert 
+   done = TodoItem("9876", "Done", "Test Title", date.today)
+   items=[done]
+   view_model = ViewModel(items=items)
+   
+   
+   assert len(view_model.done_items) == 1
+   assert(view_model.show_done_items == True)
+
+def test_view_recent_done_items():
+   # assert 
+   done = TodoItem("9876", "Done", "Test Title", date.today())
+   items=[done]
+   view_model = ViewModel(items=items) 
+  
+   assert len(view_model.recent_done_items) == 1
+
+def test_view_old_done_items():
+   # assert 
+   done = TodoItem("9876", "Done", "Test Title", date.today()-timedelta(1))
+   items=[done]
+   view_model = ViewModel(items=items) 
+  
+   assert len(view_model.older_done_items) == 1 
```

Comment:
Your code doesn't actually work, and these tests don't catch the issue.

- Your code compares the Trello data's `dateLastActivity` to `datetime.date.today()`. An example of dateLastActivity is `"2021-03-15T14:43:39.105Z"`. That is a string so you cannot compare it to a date object. Also, it contains time so even if you could, equality would fail. You should parse the string inside `from_trello_card` and save the resulting object on self, rather than just a string. The app works because your app doesn't actually use `older_done_items` or `recent_done_items`
- The test uses the TodoItem init function rather than TodoItem.from_trello_card so it doesn't catch the issue
- Also, you are only adding one item to the `items` list, so if `older_done_items` and `recent_done_items` simply returned everything, both tests would still pass.